R/functions.R

```diff
@@ -4413,3 +4411,140 @@ verifyNumeric <- function(variableToCheck) {
 chmTreeGaps <- function(numberOfCuts) {
   return(new(Class = "treeCuts", numberOfCuts = as.integer(numberOfCuts)))
 }
+
+#' Generate Default Panel Layout
+#'
+#' This function generates a default panel layout for a map.
+#' It creates two panes with equal width and height, and places them in a container.
+#' The container is then placed in a panel layout.
+#'
+#' @return A panel layout containing a container, which in turn contains two panes.
+#' @noRd
+#' @keywords internal
+default_panel_layout <- function() {
+  pane1 <- pane(
+    id = "pane1",
+    width = 50,
+    height = 100,
+    collapsed = FALSE,
+    expanded = FALSE
+  )
+  pane2 <- pane(
+    id = "pane2",
+    width = 50,
+    height = 100,
+    collapsed = FALSE,
+    expanded = FALSE
+  )
+  container1 <- container(
+    id = "ngChmContainer1",
+    width = 100,
+    height = 100,
+    vertical = FALSE,
+    children = list(pane1, pane2)
+  )
+  panel_layout <- panel_layout(
+    children = container1
+  )
+}
+
+#' Generate default Three-Panel Layout
+#'
+#' This function generates a layout with three panels for a map. The geometry is:
+#'
+#' \preformatted{
+#'          | pane2
+#'    pane1 | ------
+#'          | pane3
+#'    }
+#'
+#' @return A panel layout containing a container with three panes: two detail panes and one summary pane.
+#'
+#' @examples
+#' # Example usage:
+#' layout <- three_panel_layout()
+#' print(layout)
+#'
+#' @export
+three_panel_layout <- function() {
+  pane1 <- pane(
+    id = "pane1",
+    width = 50,
+    height = 100,
+    collapsed = FALSE,
+    expanded = FALSE
+  )
+  pane2 <- pane(
+    id = "pane2",
+    width = 100,
+    height = 50,
+    collapsed = FALSE,
+    expanded = FALSE
+  )
+  pane3 <- pane(
+    id = "pane3",
+    width = 100,
+    height = 50,
+    collapsed = FALSE,
+    expanded = FALSE
+  )
+  container2 <- container(
+    id = "ngChmContainer2",
+    width = 100,
+    height = 100,
+    vertical = TRUE,
+    children = list(pane2, pane3)
+  )
+  container1 <- container(
+    id = "ngChmContainer1",
+    width = 100,
+    height = 100,
+    vertical = FALSE,
+    children = list(pane1, container2)
+  )
+  panel_layout <- panel_layout(
```

Comment:
From @bmbroom : `panel_layout <- list("pane1", list("pane2", "pane3"))`

With the goal of this being interpreted as a top-level horizontal container, with a pane1 on the left, and a vertical container on the right, with pane2 above pane3. (same as example in code).